archivebox/extractors/__init__.py

```diff
@@ -136,24 +128,34 @@ def archive_link(link: Link, overwrite: bool=False, methods: Optional[Iterable[s
 
     return link
 
-
 @enforce_types
-def archive_links(links: List[Link], overwrite: bool=False, methods: Optional[Iterable[str]]=None, out_dir: Optional[str]=None) -> List[Link]:
-    if not links:
+def archive_links(all_links: any, overwrite: bool=False, methods: Optional[Iterable[str]]=None, out_dir: Optional[str]=None) -> List[Link]:
```

Comment:
```suggestion
def archive_links(all_links: Union[Iterable[Link], QuerySet[Snapshot]], overwrite: bool=False, methods: Optional[Iterable[str]]=None, out_dir: Optional[str]=None) -> List[Link]:
```

Reply:
How can I load? Importing Snapshot breaks because django has not been initialized. Unless I manually initialize it causes issues, and I am not a fan of that either, because I would need to make sure that wherever this function is called, django is initialized.

Reply:
The reason we cannot run `django_setup` at the entrypoint is because some commands can be run before the index is initialized, e.g. `archivebox help`, `archivebox version`, `archivebox init`.

That being said, this function can *only* be called once django is already initialized, so it's safe to depend on django models within it. Where is this function currently being called that doesn't have django already initialized? That callsite should probably be fixed.

Reply:
Yes, it is raising the issue. I will check it then.

Reply:
@pirate the issue is that `Snapshot` needs to be imported at the module level, and this module is imported at the beginning of `main.py`. Can we bypass the check for now?

Reply:
Can you use the `if False: import ...` trick to only import it for mypy?